src/components/Conference.tsx

```diff
@@ -429,33 +430,40 @@ export class Conference extends React.Component<IConferenceProps, IConferenceSta
                 screenCaptureConstraints.video.mandatory.chromeMediaSourceId = sourceId;
                 return navigator.mediaDevices.getUserMedia(screenCaptureConstraints as any);
             }).then(stream => {
-                if (this.localStream && this.localStream.getAudioTracks().length > 0) {
+                if (this.localCamStream && this.localCamStream.getAudioTracks().length > 0) {
                     // NOTE(gaolw): Merge the audio track into the screen capture stream.
-                    stream.addTrack(this.localStream.getAudioTracks()[0]);
+                    stream.addTrack(this.localCamStream.getAudioTracks()[0]);
                 }
 
-                this.gotStream(stream, {
+                stream.getVideoTracks()[0].onended = this.onScreenMediaEnded;
+                this.setLocalStream(stream, {
                     isScreenSharing: true
                 });
+
             })
             .catch(this.handleMediaException);;
     }
 
+    private onScreenMediaEnded(e: any) {
+        // NOTE(gaolw): switch back to web cam media.
+        this.getUserMedia();
+    }
+
     private handleMediaException(error: MediaStreamError) {
         // Exception type list: https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia
         this.onError(createConferenceErrorGetUserMedia(error));
     }
 
-    private gotStream(stream: MediaStream, conferenceStream?: Partial<ConferenceStream>) {
+    private setLocalStream(stream: MediaStream, conferenceStream?: Partial<ConferenceStream>) {
         const oldStream = this.state.localStream ? this.state.localStream.stream : null;
         this.setState({
             localStream: {
                 ...this.state.localStream,
                 ...conferenceStream,
                 stream,
                 local: true,
-                audioEnabled: true,
-                videoEnabled: true,
+                audioEnabled: this.state.localStream.audioEnabled,
```

Comment:
I think you can remove these two since they are deconstructed above

Reply:
done